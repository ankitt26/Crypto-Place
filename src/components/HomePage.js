import { Link } from 'react-router-dom';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { FaSpinner } from 'react-icons/fa';
import { FetchData } from '../redux/Coins/CoinSlice';
import styles from '../CSS/HomePage.module.css';
import Navbar from './NavBar';

const HomePage = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.Coins);
  const SearchKey = useSelector((state) => state.Search.searchkey);
  const { CoinList, Loading, Error } = data;

  let finalData = CoinList;

  useEffect(() => {
    if (CoinList.length === 0) {
      dispatch(FetchData());
    }
  }, [CoinList.length, dispatch]);

  if (SearchKey !== '') {
    finalData = CoinList.filter((element) => (
      element.id.includes(SearchKey)
        || element.name.includes(SearchKey)
        || element.symbol.includes(SearchKey)
    ));
  }

  // render on conditions

  const conditions = () => {
    if (Error === true) {
      return (
        <>
          <h1 className={styles.warn}>⚠️</h1>
          <h2 className={styles.errorM}> Something went wrong... </h2>
        </>
      );
    } if (Loading === true) {
      return (
        <span>
          <FaSpinner className={styles.Loading} id={styles.Loading} />
        </span>
      );
    } if (finalData.length === 0) {
      return (
        <h2 className={styles.noMatch}>
          ☹️ No Coins Found!
          {' '}
          <br />
          {' '}
          Search for another coin or try again
          later......
          {' '}
          <br />
        </h2>
      );
    }
    return null;
  };

  // rendering the map data

  const renderCoins = finalData.map((element) => (
    <Link className={styles.card} to={`/coin/${element.id}`} key={element.id}>
      <img className={styles.img} src={element.image} alt="coin" />
      <div className={styles.detail_box}>
        <div className={styles.box1}>
          <h4 className={styles.name}>Coin</h4>
          <h5 className={styles.symbol}>Symbol</h5>
          <h5 className={styles.price}>Current Price</h5>
          <h5 className={styles.change}>24h Change</h5>
        </div>
        <div className={styles.box2}>
          <h4 className={styles.name}>{element.name}</h4>
          <h5 className={styles.symbol}>{element.symbol}</h5>
          <h5 className={styles.price}>
            $
            {element.current_price}
          </h5>
          <h5 className={styles.change}>
            $
            {' '}
            {element.price_change_percentage_24h}
          </h5>
        </div>
      </div>
    </Link>
  ));

  // final return

  return (
    <>
      <Navbar />
      <div>{conditions()}</div>
      <main className={styles.main}>{renderCoins}</main>
    </>
  );
};

export default HomePage;
