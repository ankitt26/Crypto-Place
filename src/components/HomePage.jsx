import Navbar from "./NavBar";
import { Link } from "react-router-dom";
import styles from "../CSS/HomePage.module.css";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { FetchData } from "../redux/Coins/CoinSlice";
import { FaSpinner } from "react-icons/fa";

const HomePage = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.Coins);
  const SearchKey = useSelector((state) => state.Search.searchkey);
  const { CoinList, Loading, Error } = data;

  let finalData = CoinList;

  useEffect(() => {
    if (CoinList.length === 0) {
      dispatch(FetchData());
    }
  }, [dispatch]);

  if (SearchKey !== "") {
    finalData = CoinList.filter((element) => {
      return (
        element.id.includes(SearchKey) ||
        element.name.includes(SearchKey) ||
        element.symbol.includes(SearchKey)
      );
    });
  }

  // render on conditions

  const conditions = () => {
    if (Error === true) {
      return (
        <>
          <h1 className={styles.warn}>⚠️</h1>
          <h2 className={styles.errorM}> Something went wrong... </h2>
        </>
      );
    } else if (Loading === true) {
      return (
        <span>
          <FaSpinner className={styles.Loading} id={styles.Loading} />
        </span>
      );
    } else if (finalData.length === 0) {
      return (
        <h2 className={styles.noMatch}>
          ☹️ No Coins Found! <br /> Search for another coin or try again
          later...... <br />
        </h2>
      );
    }
  };

  // rendering the map data

  const renderCoins = finalData.map((element) => {
    return (
      <Link className={styles.card} to={`/coin/${element.id}`} key={element.id}>
        <img className={styles.img} src={element.image}></img>
        <div className={styles.detail_box}>
          <div className={styles.box1}>
            <h4 className={styles.name}>Coin</h4>
            <h5 className={styles.symbol}>Symbol</h5>
            <h5 className={styles.price}>Current Price</h5>
            <h5 className={styles.change}>24h Change</h5>
          </div>
          <div className={styles.box2}>
            <h4 className={styles.name}>{element.name}</h4>
            <h5 className={styles.symbol}>{element.symbol}</h5>
            <h5 className={styles.price}>$ {element.current_price}</h5>
            <h5 className={styles.change}>
             $ {element.price_change_percentage_24h}
            </h5>
          </div>
        </div>
      </Link>
    );
  });

  // final return

  return (
    <>
      <Navbar />
      <div>{conditions()}</div>
      <main className={styles.main}>{renderCoins}</main>
    </>
  );
};

export default HomePage;
