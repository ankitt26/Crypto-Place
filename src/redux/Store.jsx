import { configureStore } from '@reduxjs/toolkit';
import CoinSlice from './Coins/CoinSlice';
import SearchSlice from './Coins/SearchSlice';
const Store = configureStore({
    reducer: {
        Coins : CoinSlice,
        Search : SearchSlice
    },
})

export default Store ;