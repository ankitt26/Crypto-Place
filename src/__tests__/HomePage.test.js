import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomePage from '../components/HomePage';
import Store from '../redux/Store';

it('homepage test', () => {
  const component = renderer.create(
    <Provider store={Store}>
      <HomePage />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
