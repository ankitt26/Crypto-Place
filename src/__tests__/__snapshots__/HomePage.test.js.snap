// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage test 1`] = `
Array [
  <nav
    className="nav"
  >
    <h2
      className="h2"
    >
      CryptoPlace
    </h2>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="   search"
        value=""
      />
      <button
        className="btn"
        type="submit"
      >
        <svg
          className="search"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.504 13.616l-3.79-3.223c-0.392-0.353-0.811-0.514-1.149-0.499 0.895-1.048 1.435-2.407 1.435-3.893 0-3.314-2.686-6-6-6s-6 2.686-6 6 2.686 6 6 6c1.486 0 2.845-0.54 3.893-1.435-0.016 0.338 0.146 0.757 0.499 1.149l3.223 3.79c0.552 0.613 1.453 0.665 2.003 0.115s0.498-1.452-0.115-2.003zM6 10c-2.209 0-4-1.791-4-4s1.791-4 4-4 4 1.791 4 4-1.791 4-4 4z"
          />
        </svg>
      </button>
    </form>
  </nav>,
  <div>
    <span>
      <svg
        className="Loading"
        fill="currentColor"
        height="1em"
        id="Loading"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
        />
      </svg>
    </span>
  </div>,
  <main
    className="main"
  />,
]
`;
